sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs

```diff
@@ -135,5 +183,215 @@ public override async ValueTask<AccessToken> GetTokenAsync(TokenRequestContext r
+                        using (Stream s = File.OpenRead(clientCertificatePath))
+                        {
+                            while (true)
+                            {
+                                int read = await s.ReadAsync(buf, offset, buf.Length, cancellationToken).ConfigureAwait(false);
+                                for (int i = 0; i < read; i++)
+                                {
+                                    certContents.Add(buf[i]);
+                                }
+
+                                if (read == 0)
+                                {
+                                    break;
+                                }
+                            }
+                        }
+
+                        Certificate = new X509Certificate2(certContents.ToArray());
+                    }
+
+                    return Certificate;
+                }
+                catch (Exception e) when (!(e is OperationCanceledException))
+                {
+                    throw new CredentialUnavailableException("Could not load certificate file", e);
+                }
+            }
+
+            private async Task<X509Certificate2> LoadCertificateFromPemFileAsync(bool async, string clientCertificatePath, CancellationToken cancellationToken)
+            {
+                if (!(Certificate is null))
+                {
+                    return Certificate;
+                }
+
+                string certficateText;
+
+                try
+                {
+                    if (!async)
+                    {
+                        certficateText = File.ReadAllText(clientCertificatePath);
+                    }
+                    else
+                    {
+                        StringBuilder sb = new StringBuilder();
+                        char[] buf = new char[BufferSize];
+                        int offset = 0;
+
+                        using (StreamReader sr = new StreamReader(clientCertificatePath))
+                        {
+                            while (true)
+                            {
+                                cancellationToken.ThrowIfCancellationRequested();
+                                int read = await sr.ReadAsync(buf, offset, buf.Length).ConfigureAwait(false);
+
+                                for (int i = 0; i < read; i++)
+                                {
+                                    sb.Append(buf[i]);
+                                }
+
+                                if (read == 0)
+                                {
+                                    break;
+                                }
+                            }
+
+                            certficateText = sb.ToString();
+                        }
+                    }
+
+                    Regex certificateRegex = new Regex("(-+BEGIN CERTIFICATE-+)(\n\r?|\r\n?)([A-Za-z0-9+/\n\r]+=*)(\n\r?|\r\n?)(-+END CERTIFICATE-+)", RegexOptions.CultureInvariant);
```

Comment:
`(\n\r?|\r\n?)` -> `[\r\n]+` ?

Reply:
Took this RegEx from our [TypeScript Identity SDK](https://github.com/Azure/azure-sdk-for-js/blob/9d5747196fbae2c6006402f343c784cb796ee24c/sdk/identity/identity/src/credentials/clientCertificateCredential.ts), I'd prefer to leave it as is.

Reply:
Ah, sounds good.